packages/gossip/manager.go

```diff
@@ -3,23 +3,32 @@ package gossip
 import (
 	"fmt"
 	"net"
+	"runtime"
 	"sync"
 
 	"github.com/golang/protobuf/proto"
 	"github.com/iotaledger/goshimmer/packages/binary/messagelayer/message"
 	pb "github.com/iotaledger/goshimmer/packages/gossip/proto"
 	"github.com/iotaledger/goshimmer/packages/gossip/server"
-	"github.com/iotaledger/hive.go/async"
 	"github.com/iotaledger/hive.go/autopeering/peer"
 	"github.com/iotaledger/hive.go/events"
 	"github.com/iotaledger/hive.go/identity"
 	"github.com/iotaledger/hive.go/logger"
+	"github.com/iotaledger/hive.go/workerpool"
 )
 
 const (
 	maxPacketSize = 64 * 1024
 )
 
+var (
+	messageWorkerCount     = runtime.GOMAXPROCS(0) * 4
+	messageWorkerQueueSize = 1000
+
+	messageRequestWorkerCount     = 1
```

Comment:
I was thinking it might be smarter to have more than 1 worker here since a message might be loaded from disk which would block the whole thing for that time.